audformat/core/database.py

```diff
@@ -289,6 +291,40 @@ def drop_tables(
         for table_id in table_ids:
             self.tables.pop(table_id)
 
+    def file_duration(
+            self,
+            file: str = None,
+    ) -> pd.Timedelta:
+        r"""Duration of a single file or all files in the database.
+
+        Args:
+            file: if a file name is given,
+                returns duration of the file.
+                If ``None`` returns the summed duration
+                of all files in the database
+
+        Returns:
+            duration
+
+        """
+        if file is None:
+            return functools.reduce(
+                lambda d, f: d + self.file_duration(f),
+                self.files,
+                pd.to_timedelta(0),
+            )
+        elif file not in self._file_duration:
+            if not os.path.isabs(file) and self.root is not None:
+                path = os.path.join(self.root, file)
+            else:
+                path = file
+            dur = audiofile.duration(path)
+            dur = pd.to_timedelta(dur, unit='s')
+            self._file_duration[file] = dur
```

Comment:
Above you allow that `file` is an absolute path, which is ok for running `audiofile` on it. But when we then store it allowing also relative or absolute path, it means that it `self._file_duration` can get twice as big as `db.files` in the end (or even bigger if the user moves around the database in between).

Reply:
One way around this would be that we allow only the syntax that is used in the current index (e.g. `db.files`).
But again, `db.files` is generated on the fly and it will change if a user updates a table in between.

Reply:
I guess whether `file` is absolute or relative depends mainly on how the files are stored in the table. We could of course always make it relative by removing `db.root`, but I'm not sure if this really worth the effort. In the tables we also usually store absolute paths.

Reply:
I also don't think moving around a database is a common use case, is it?

Reply:
On the other hand - always storing the relative path would indeed solve the problem :)

Reply:
To not have to deal with https://github.com/audeering/audformat/pull/114#discussion_r744586885 maybe we indeed only store the relative path of the files. 

> I also don't think moving around a database is a common use case, is it?

It is if you do `db.update(db_new)`, where `db_new` is moved to the new location. Will it in general work if both `db` and `db_new` contain already entries in `db._files_duration`?

Reply:
But `db.update()` just add some new files, I don't see why this is a problem.

Reply:
The question was more what will happen with its `_files_duration` entries. Will they be merged with the ones from `db`? If yes, it will only make sense if the path is stored relative as the absolute path would now be different.

Reply:
Mhh, yes if we store the file paths relative we could indeed update `_files_duration`. Otherwise, it would simply re-calculate the duration for the new files when `files_duration()` is called.

At the moment I'm not completely convinced if we should really store the relative path, though. Cause then there is of course the question what `files_duration()` should return. Always the relative path? Always the full path? At the moment it will return it the same way it's stored in the tables. And I don't think that we will often have the case that the user requests a database with absolute path, but then calls `files_duration()` or vice versa. At least as long as the user passes `db.files` or `db[table_id].files` this is not the case. And even if the user does it, the result will still be correct, though it has re-calculate the duration in that case. But as I said, I don't see it as a common use case.

Reply:
I'm also fine with your proposal. To summarize it:

* `db.update(db_new)` will simply ignore the entries in `db_new._files_duration`
* in `db._files_duration` we store relative and absolute paths as different entries. This is not a problem as the user should not access this dictionary
* we could simply recalculate the duration if a user has called first absolute path and then relative (or vice versa), but I think we could even avoid it, see https://github.com/audeering/audformat/pull/114#discussion_r744586885

Reply:
If I see it correctly the first two points are already implemented by the current behavior, correct?

Reply:
Yes, and with the latest commit the third point should be implemented, too. See my comment at #114.